### ISSUE TITLE
device.connect gives ringing tone and immediate a disconnect tone, call does not connect.

### ISSUE THREAD

**Issue by ricksonmenezes** [twilio/twilio-video.js#1706]
I have integrated (migrated to)  the 2.1.0 VoiceSDK (JS) version of Twilio but I am unable to get a call to go through. 

I have changed all the old Twilio.Device calls and my code looks like this. (the vm is because we are doing the calls via vuejs)

`var tokenId = "<%=token%>";

                vm.device = new Twilio.Device(tokenId);

                const deviceOptions = {debug: false};
                vm.device.updateOptions(deviceOptions)
                vm.device.register();

                vm.device.on('error', (twilioError, call) => {
                    console.log('An error has occurred: ', + twilioError);
                });

                vm.device.on('registered', device => {
                    console.log('    twilio registered........');
                });

                vm.device.on('registering', device => {
                    console.log('    twilio registering...    ');

            });
                vm.device.on('tokenWillExpire', () => {
               //     const token = getNewTokenViaAjax(); call API to get new token
                    console.log('twilio token expired')
                //device.updateToken(token);
            });
                vm.device.on('unregistered', () => {
                console.log("    twilio unregistered...    ");
            });

                vm.device.on('incoming', conn => {
                console.log("Incoming connection from " + conn.parameters.From);
                conn.accept()
            });

                vm.device.on('connect', connection => {
                    console.log("connection opened " + connection);
                });

                vm.device.on('disconnect', connection => {
                    console.log("connection disonnected " + connection);

                });`

I am able to see the "device registered"  in the console logs to which I assume the token received from the server works fine and twilio registers the device for incoming calls. The twilio version on the server is twilio-java-sdk 3.3.9 jar using the Capability Token API 

```
TwilioCapability capability = new TwilioCapability(HIFIConstants.TWILIO_ACCOUNT_SID, HIFIConstants.TWILIO_AUTH_TOKEN)
capability.allowClientIncoming(HIFIConstants.TWILIO_CLIENT_INCOMING);
 capability.allowClientOutgoing(applicationSid);
token = capability.generateToken();
```

The logs shown when it tries to connect are the following. It seems to me that the logs are API endpoint calls. I am copying the payload as that is where all the information is to determine what the issue is. 

Below are the payloads that are always sent on the same enpoint below. The response to all these calls is always "OK". 

https://eventgw.us1.twilio.com/v4/EndpointEvents

{"publisher":"twilio-js-sdk","group":"connection","name":"outgoing","timestamp":"2022-02-12T05:16:37.183Z","level":"INFO","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE","preflight":false},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"network-information","name":"network-change","timestamp":"2022-02-12T05:16:37.188Z","level":"INFO","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE","downlink":10,"effective_type":"4g","rtt":50},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"get-user-media","name":"succeeded","timestamp":"2022-02-12T05:16:37.807Z","level":"INFO","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE","data":{"audioConstraints":{"audio":true}}},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"signaling-state","name":"have-local-offer","timestamp":"2022-02-12T05:16:37.854Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"dtls-transport-state","name":"new","timestamp":"2022-02-12T05:16:37.859Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"ice-gathering-state","name":"gathering","timestamp":"2022-02-12T05:16:37.865Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"ice-candidate","name":"ice-candidate","timestamp":"2022-02-12T05:16:37.873Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE","candidate_type":"host","deleted":false,"ip":"172.27.32.1","is_remote":false,"port":51986,"priority":2122260223,"protocol":"udp","related_address":null,"related_port":null,"tcp_type":null,"transport_id":"0"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"ice-candidate","name":"ice-candidate","timestamp":"2022-02-12T05:16:37.878Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE","candidate_type":"host","deleted":false,"ip":"192.168.1.2","is_remote":false,"network-cost":10,"port":51987,"priority":2122194687,"protocol":"udp","related_address":null,"related_port":null,"tcp_type":null,"transport_id":"0"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"ice-candidate","name":"ice-candidate","timestamp":"2022-02-12T05:16:37.967Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE","candidate_type":"host","deleted":false,"ip":"172.27.32.1","is_remote":false,"port":9,"priority":1518280447,"protocol":"tcp","related_address":null,"related_port":null,"tcp_type":"active","transport_id":"0"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"ice-candidate","name":"ice-candidate","timestamp":"2022-02-12T05:16:37.971Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE","candidate_type":"host","deleted":false,"ip":"192.168.1.2","is_remote":false,"network-cost":10,"port":9,"priority":1518214911,"protocol":"tcp","related_address":null,"related_port":null,"tcp_type":"active","transport_id":"0"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"ice-gathering-state","name":"complete","timestamp":"2022-02-12T05:16:37.975Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"connection","name":"outgoing-ringing","timestamp":"2022-02-12T05:16:38.131Z","level":"INFO","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE","hasEarlyMedia":false},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"ice-connection-state","name":"checking","timestamp":"2022-02-12T05:16:38.555Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"settings","name":"edge","timestamp":"2022-02-12T05:16:38.559Z","level":"INFO","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE","edge":"singapore"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"signaling-state","name":"stable","timestamp":"2022-02-12T05:16:38.561Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"connection","name":"accepted-by-remote","timestamp":"2022-02-12T05:16:38.568Z","level":"INFO","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"settings","name":"codec","timestamp":"2022-02-12T05:16:38.573Z","level":"INFO","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE","codec_params":"","selected_codec":"PCMU/8000"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"pc-connection-state","name":"connecting","timestamp":"2022-02-12T05:16:38.608Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"ice-candidate","name":"selected-ice-candidate-pair","timestamp":"2022-02-12T05:16:38.665Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE","local_candidate":{"candidate_type":"prflx","deleted":false,"ip":"223.229.188.54","is_remote":false,"network-cost":10,"port":51987,"priority":1853759231,"protocol":"udp","related_address":"192.168.1.2","related_port":51987,"tcp_type":null,"transport_id":""},"remote_candidate":{"candidate_type":"host","deleted":false,"ip":"3.1.77.190","is_remote":true,"port":15920,"priority":2130706431,"protocol":"udp","related_address":null,"related_port":null,"tcp_type":null,"transport_id":""}},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"dtls-transport-state","name":"connecting","timestamp":"2022-02-12T05:16:38.668Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"ice-connection-state","name":"connected","timestamp":"2022-02-12T05:16:38.671Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"dtls-transport-state","name":"connected","timestamp":"2022-02-12T05:16:38.832Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"pc-connection-state","name":"connected","timestamp":"2022-02-12T05:16:38.835Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"connection","name":"disconnected-by-remote","timestamp":"2022-02-12T05:16:39.552Z","level":"INFO","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

{"publisher":"twilio-js-sdk","group":"dtls-transport-state","name":"closed","timestamp":"2022-02-12T05:16:39.564Z","level":"DEBUG","payload_type":"application/json","private":false,"payload":{"aggressive_nomination":false,"browser_extension":false,"dscp":true,"ice_restart_enabled":true,"platform":"WebRTC","sdk_version":"2.1.0","call_sid":"CAf86f583f7562b1e0c38050597d478d57","temp_call_sid":"TJS52891adc-f5d7-4fad-804f-de67c42d0a1e","direction":"OUTGOING","region":"ASIAPAC_SINGAPORE"},"publisher_metadata":{}}

What I see in the last two logs is that it is disconnected by remote and then closes. I am not disconnecting the call myself. 

Also the twilio signal URL 

wss://chunderw-vpc-gll.twilio.com/signal

that is called by the library stays on PENDING on the chrome network inspect for a long time and then moves to status UPGRADE. It's status is always 101 (switching?) but it never becomes 200. Don't know if this issue is related to the call not being able to connect.. 

I have been reading up on the scp paragraph in the [github page](https://github.com/twilio/twilio-voice.js)  It talks of configuring SCP so that the signal URL is allowed to go through. Given the logs I have above that is able to register the device, I am wondering if I need to add this scp tag on the page. In fact, I did try it but it gave me other errors like "CSP tag cannot be used in meta please use header" and "Refused to load the script because it violates the following Content Security Policy directive". This is possible as I am not using the write syntax but just wanted to mention what I have tried. 

I don't get any other logs in the browser except the endpoint logs. The logs on twilio account website don't show up for the number I call to which I suppose is because the call has not gone through. The number I call is a US number and I have tried calling the number adding country code 1 and +1 and both fail. They fail with ringing tone and disconnect tone immediately after. 

 - [ ] I have verified that the issue occurs with the latest twilio-video.js release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-video.js/blob/master/CHANGELOG.md).
 - [ ] I reviewed the [Common Issues](https://github.com/twilio/twilio-video.js/blob/master/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
 - [ ] I verified that the [Quickstart](https://github.com/twilio/video-quickstart-js) application works in my environment.
 - [ ] I am not sharing any [Personally Identifiable Information (PII)](https://www.twilio.com/docs/glossary/what-is-personally-identifiable-information-pii)
       or sensitive account information (API keys, credentials, etc.) when reporting this issue.

**Code to reproduce the issue:**

If possible, please include debug logs. You can set the log level for
twilio-video.js by calling `connect` with the `logLevel` property, e.g.

    connect(token, { logLevel: 'debug' })

Then you can copy and paste or save the logs from your browser's console.

-->

TODO

**Software versions:**

 - [ ] Browser(s): chrome 97
 - [ ] Operating System: Windows 10
 - [ ] twilio-video.js: 2.1 using it as a single JS file from /dist folder as mentioned in the migration documentation. 
 - [ ] Third-party libraries:VueJS

**Comment by manjeshbhargav** [comment]
Hi @ricksonmenezes ,

This repository is for the Video SDK. Let me move your issue to the Voice SDK.

Thanks,

Manjesh
